src/file.ts

```diff
@@ -2280,6 +2432,195 @@ class File extends ServiceObject<File> {
+   */
+  /**
+   * @typedef {array} GetSignedPolicyV4Response
+   * @property {SignedPolicyV4Output} 0 An object containing the request URL and form fields.
+   */
+  /**
+   * @callback GetSignedPolicyV4Callback
+   * @param {?Error} err Request error, if any.
+   * @param {SignedPolicyV4Output} output An object containing the request URL and form fields.
+   */
+  /**
+   * Get a v4 signed policy document to allow a user to upload data with a POST
+   * request.
+   *
+   * In Google Cloud Platform environments, such as Cloud Functions and App
+   * Engine, you usually don't provide a `keyFilename` or `credentials` during
+   * instantiation. In those environments, we call the
+   * [signBlob
+   * API](https://cloud.google.com/iam/reference/rest/v1/projects.serviceAccounts/signBlob#authorization-scopes)
+   * to create a signed policy. That API requires either the
+   * `https://www.googleapis.com/auth/iam` or
+   * `https://www.googleapis.com/auth/cloud-platform` scope, so be sure they are
+   * enabled.
+   *
+   * @see [Policy Document Reference]{@link https://cloud.google.com/storage/docs/xml-api/post-object#policydocument}
+   *
+   * @param {object} options Configuration options.
+   * @param {Date|number|string} options.expires - A timestamp when this policy will expire. Any
+   *     value given is passed to `new Date()`.
+   * @param {boolean} [config.virtualHostedStyle=false] Use virtual hosted-style
+   *     URLs ('https://mybucket.storage.googleapis.com/...') instead of path-style
+   *     ('https://storage.googleapis.com/mybucket/...'). Virtual hosted-style URLs
+   *     should generally be preferred instaed of path-style URL.
+   *     Currently defaults to `false` for path-style, although this may change in a
+   *     future major-version release.
+   * @param {string} [config.bucketBoundHostname] The bucket-bound hostname to return in
+   *     the result, e.g. "https://cdn.example.com".
+   * @param {object} [config.fields] [Form fields]{@link https://cloud.google.com/storage/docs/xml-api/post-object#policydocument}
+   *     to include in the signed policy. Any fields with key beginning with 'x-ignore-'
+   *     will not be included in the policy to be signed.
+   * @param {object[]} [config.conditions] [Conditions]{@link https://cloud.google.com/storage/docs/authentication/signatures#policy-document}
+   *     to include in the signed policy. All fields given in `config.fields` are
+   *     automatically included in the conditions array, adding the same entry
+   *     in both `fields` and `conditions` will result in duplicate entries.
+   *
+   * @param {GetSignedPolicyV4Callback} [callback] Callback function.
+   * @returns {Promise<GetSignedPolicyV4Response>}
+   *
+   * @example
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const myBucket = storage.bucket('my-bucket');
+   *
+   * const file = myBucket.file('my-file');
```

Comment:
Writing canonicals, I almost want to say you should move up `getSignedPolicyV4()` to bucket instead of file.